src/make_watertight/Gen.cpp

```diff
@@ -0,0 +1,2117 @@
+        }
+      }
+    }
+  }
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Gen::squared_dist_between_verts(const moab::EntityHandle v0,
+                                                const moab::EntityHandle v1,
+                                                double& d) {
+  moab::ErrorCode result;
+  moab::CartVect coords0, coords1;
+  result = MBI()->get_coords(&v0, 1, coords0.array());
+  if (moab::MB_SUCCESS != result) {
+    std::cout << "dist_between_verts: get_coords on v0=" << v0 << " result="
+              << result << std::endl;
+    return result;
+  }
+  result = MBI()->get_coords(&v1, 1, coords1.array());
+  if (moab::MB_SUCCESS != result) {
+    std::cout << "dist_between_verts: get_coords on v1=" << v1 << " result="
+              << result << std::endl;
+    return result;
+  }
+  const moab::CartVect diff = coords0 - coords1;
+  d = diff.length_squared();
+  return moab::MB_SUCCESS;
+}
+
+double Gen::dist_between_verts(const moab::CartVect v0, const moab::CartVect v1) {
+  moab::CartVect v2 = v0 - v1;
+  return v2.length();
+}
+moab::ErrorCode Gen::dist_between_verts(const moab::EntityHandle v0, const moab::EntityHandle v1, double& d) {
+  moab::ErrorCode result;
+  moab::CartVect coords0, coords1;
+  result = MBI()->get_coords(&v0, 1, coords0.array());
+  if (moab::MB_SUCCESS != result) {
+    std::cout << "dist_between_verts: get_coords on v0=" << v0 << " result="
+              << result << std::endl;
+    return result;
+  }
+  result = MBI()->get_coords(&v1, 1, coords1.array());
+  if (moab::MB_SUCCESS != result) {
+    std::cout << "dist_between_verts: get_coords on v1=" << v1 << " result="
+              << result << std::endl;
+    return result;
+  }
+  d = dist_between_verts(coords0, coords1);
+  return moab::MB_SUCCESS;
+}
+
+double Gen::dist_between_verts(double coords0[], double coords1[]) {
+  return sqrt((coords0[0] - coords1[0]) * (coords0[0] - coords1[0]) +
+              (coords0[1] - coords1[1]) * (coords0[1] - coords1[1]) +
+              (coords0[2] - coords1[2]) * (coords0[2] - coords1[2]));
+}
+double Gen::dist_between_verts(moab::EntityHandle vert0, moab::EntityHandle vert1) {
+  double coords0[3], coords1[3];
+  moab::ErrorCode result;
+  result = MBI()->get_coords(&vert0, 1, coords0);
+  if (moab::MB_SUCCESS != result)
+    std::cout << "result=" << result << " vert="
+              << vert0 << std::endl;
+  assert(moab::MB_SUCCESS == result);
+  result = MBI()->get_coords(&vert1, 1, coords1);
+  if (moab::MB_SUCCESS != result)
+    std::cout << "result=" << result << " vert="
+              << vert1 << std::endl;
+  assert(moab::MB_SUCCESS == result);
+  return dist_between_verts(coords0, coords1);
+}
+
+// Return the length of the curve defined by moab::MBEDGEs or ordered moab::MBVERTEXs.
+double Gen::length(std::vector<moab::EntityHandle> edges) {
+  if (edges.empty())
+    return 0;
+
+  moab::ErrorCode result;
+  std::vector<moab::EntityHandle>::iterator i;
```

Comment:
This variable may be unused.  It's shadowed by `i` declared on line 438, and `i` declared on 466.